core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+        import dataAccess.multiregionalBucketMigrationQuery._
+        (for {
+          migration <- getAttempt(migrationId)
+          _ <- OptionT.liftF {
+            update5(
+              migrationId,
+              finishedCol,
+              Option.empty[Timestamp],
+              outcomeCol,
+              Option.empty[String],
+              messageCol,
+              Option.empty[String],
+              // reconfigure sts iam permissions for good measure
+              if (migration.tmpBucketTransferIamConfigured.isDefined) tmpBucketTransferIamConfiguredCol
+              else workspaceBucketTransferIamConfiguredCol,
+              Option.empty[Timestamp],
+              // re-issue the job
+              if (migration.tmpBucketTransferIamConfigured.isDefined) tmpBucketTransferJobIssuedCol
+              else workspaceBucketTransferJobIssuedCol,
+              Option.empty[Timestamp]
+            )
+          }
+        } yield ()).value
+      },
+      FailureModes.noObjectPermissionsFailure
+    )
+
+  def retryFailuresLike(update: (DataAccess, Long) => ReadWriteAction[Any],
+                        failureMessage: String,
+                        others: String*
+  ): MigrateAction[Unit] =
+    for {
+      (maxAttempts, maxRetries) <- asks(d => (d.maxConcurrentAttempts, d.maxRetries))
+      now <- nowTimestamp
+      (migrationId, workspaceName, retryCount) <- inTransactionT { dataAccess =>
+        import dataAccess.{multiregionalBucketMigrationRetryQuery, multiregionalBucketMigrationQuery}
+        for {
+          (migrationId, workspaceName) <- multiregionalBucketMigrationRetryQuery.nextFailureLike(
+            maxRetries,
+            failureMessage,
+            others: _*
+          )
+          numAttempts <- OptionT.liftF(multiregionalBucketMigrationQuery.getNumActiveResourceLimitedMigrations)
+          if numAttempts < maxAttempts
+          retryCount <- OptionT.liftF[ReadWriteAction, Long] {
+            for {
+              MultiregionalBucketMigrationRetry(id, _, numRetries) <- multiregionalBucketMigrationRetryQuery
+                .getOrCreate(migrationId)
+              retryCount = numRetries + 1
+              _ <- multiregionalBucketMigrationRetryQuery.update(id,
+                                                                 multiregionalBucketMigrationRetryQuery.retriesCol,
+                                                                 retryCount
+              )
+              _ <- update(dataAccess, migrationId)
+            } yield retryCount
+          }
+        } yield (migrationId, workspaceName, retryCount)
+      }
+      _ <- getLogger[MigrateAction].info(
+        stringify(
+          "migrationId" -> migrationId,
+          "workspace" -> workspaceName,
+          "retry" -> retryCount,
+          "retried" -> now
+        )
+      )
+    } yield ()
+
+  // Create bucket in us-central1-a
+  final def createBucketInTargetRegion(migration: MultiregionalBucketMigration,
```

Comment:
This has been changed to not take a source and destination Google project at this migration works within a single Google project. Bucket location is always `us-central1` at the moment as defined by `defaultBucketLocation`, but we could amend that in the future if we wanted to support other regions